File: agent/src/handlers/file.py
import os
import logging
from src.constants.logs import *

class FileHandler:
    def __init__(self, storage_dir: str): 
        self.storage_dir: str = storage_dir

        logging.info(STORAGE_DIR_SET.format(storage_dir))

    def _prepare_file(self, file_name: str):
        return os.path.join(self.storage_dir, file_name)

    def save_file(self, file_name: str, content: str) -> str:
        logging.info(SAVING_FILE.format(file_name))

        file_path: str = self._prepare_file(file_name)

        with open(file_path, 'w') as f:
            f.write(content)
        return file_path

    def read_file(self, file_name: str) -> str:
        logging.info(READING_FILE.format(file_name))

        content: str = None
        file_path: str = self._prepare_file(file_name)

        with open(file_path, 'r') as f:
            content = f.read()
        return content

File: agent/src/file.rs
use std::fs;
use std::io;
use log::info;

pub struct File {
    pub path: String,
}

impl File {
    pub fn new(path: &str) -> Self {
        Self { path: path.to_string() }
    }
}

pub struct FileHandler {
    pub storage_dir: String,
}

impl FileHandler {
    pub fn new(storage_dir: &str) -> Self {
        info!("Storage directory set: {}", storage_dir);
        Self { storage_dir: storage_dir.to_string() }
    }

    pub fn prepare_file(&self, file_name: &str) -> String {
        format!("{}/{}", self.storage_dir, file_name)
    }

    pub fn save_file(&self, file_name: &str, content: &str) -> io::Result<String> {
        let file_path = self.prepare_file(file_name);
        info!("Saving file: {}", file_name);
        fs::write(&file_path, content)?;
        Ok(file_path)
    }

    pub fn read_file(&self, file_name: &str) -> io::Result<String> {
        let file_path = self.prepare_file(file_name);
        info!("Reading file: {}", file_name);
        fs::read_to_string(&file_path)
    }
}
